### DISCUSSION TITLE
What is the use of the generated ids on the data-test attributes of list items.

### DISCUSSION THREAD

**Discussion by FrankAmsterdam** [cypress-io/cypress-realworld-app#1087] [General] [answered]
I noticed that on the data-test attributes for all list items like for instance '**transaction-sender-XSWbSVnsp**' the last part exists of a generated id. Is this id useful, in other words is a non-unique id not sufficient?

**Comment by kevinold** [comment, marked as answer]
@FrankAmsterdam The transaction id is useful on the end of the data-test attribute for various situations where you might want to select all elements of a certain type and/or filter them by a certain id, for instance.  This is an advanced concept, but we use it [in a test](https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/tests/ui/transaction-feeds.spec.ts#L105-L117) to find the transaction details for the transaction item we matched from the UI.
